Numsense/Portuguese.fs

```diff
@@ -0,0 +1,130 @@
+        match candidate with
+        | ""                          -> Some acc
+        | StartsWith "-"            t
+        | StartsWith "E"            t -> conv                acc  t
+        | StartsWith "ZERO"         t -> conv          (0  + acc) t
+        | StartsWith "UM"           t -> conv          (1  + acc) t
+        | StartsWith "DOIS"         t -> conv          (2  + acc) t
+        | StartsWith "TRÊS"         t -> conv          (3  + acc) t
+        | StartsWith "QUATRO"       t -> conv          (4  + acc) t
+        | StartsWith "CINCO"        t -> conv          (5  + acc) t
+        | StartsWith "SEIS"         t -> conv          (6  + acc) t
+        | StartsWith "OITO"         t -> conv          (8  + acc) t
+        | StartsWith "ONZE"         t -> conv         (11  + acc) t
+        | StartsWith "DOZE"         t -> conv         (12  + acc) t
+        | StartsWith "CATORZE"      t
+        | StartsWith "QUATORZE"     t -> conv         (14  + acc) t
+        | StartsWith "QUINZE"       t -> conv         (15  + acc) t
+        | StartsWith "DEZASSEIS"    t -> conv         (16  + acc) t
+        | StartsWith "DEZASSETE"    t -> conv         (17  + acc) t
+        | StartsWith "DEZOITO"      t -> conv         (18  + acc) t
+        | StartsWith "DEZANOVE"     t -> conv         (19  + acc) t
+        | StartsWith "DEZ"          t -> conv         (10  + acc) t
+        | StartsWith "VINTE"        t -> conv         (20  + acc) t
+        | StartsWith "TRINTA"       t -> conv         (30  + acc) t
+        | StartsWith "QUARENTA"     t -> conv         (40  + acc) t
+        | StartsWith "CINQUENTA"    t -> conv         (50  + acc) t
+        | StartsWith "SESSENTA"     t -> conv         (60  + acc) t
+        | StartsWith "SETENTA"      t -> conv         (70  + acc) t
+        | StartsWith "SETE"         t -> conv          (7  + acc) t
+        | StartsWith "OITENTA"      t -> conv         (80  + acc) t
+        | StartsWith "NOVENTA"      t -> conv         (90  + acc) t
+        | StartsWith "NOVE"         t -> conv          (9  + acc) t
+        | StartsWith "CENTOS"       t -> conv        (100 %* acc) t
+        | StartsWith "DUZENTOS"     t -> conv         (200 + acc) t
+        | StartsWith "TREZENTOS"    t -> conv         (300 + acc) t
+        | StartsWith "TREZE"        t -> conv         (13  + acc) t
+        | StartsWith "QUINHENTOS"   t -> conv         (500 + acc) t
+        | StartsWith "CENTO"        t
+        | StartsWith "CEM"          t -> conv         (100 + acc) t
+        | StartsWith "MILHÃO"       t
+        | StartsWith "MILHÕES"      t -> conv    (1000000 %* acc) t
+        | StartsWith "MILMILHÕES"   t ->
+               conv (if acc = 0 then 1000000000 else 1000000000 %* acc) t
+        | StartsWith "MIL"          t ->
+               conv (if acc = 0 then 1000 else      1000 %* acc) t
+        | _ -> failwith(candidate)
+
+    let canonicalized = x.Trim().ToUpper(System.Globalization.CultureInfo "pt-PT")
```

Comment:
Great question. Yes, this is specific to european Portuguese.

Regarding the function name, one can arguably say Portuguese implies european Portuguese and Brazilian Portuguese is used explicitly otherwise (my wife who is Brazilian would probably disagree :)).

There are some small but significant differences like the use of the long or the short scale (https://en.wikipedia.org/wiki/Long_and_short_scales) or "meia" in Brazil instead of "seis" in Portugal to denote number six.

I suspect a good alternative to the current implementation could be to rely on a CultureInfo to be passed in into a general function and rely on the strategy for each CultureInfo (one could say that in the lack of a specific culture implementation, the parent culture would handle the request). This would obviously require a significant amount of changes...

Nonetheless, changing the CultureInfo to "pt" should not have any impact since both languages use arabic numerals with a decimal comma and the same thousands separator (see https://en.wikipedia.org/wiki/Decimal_mark#Examples_of_use).

Please let me know what you think. I will ask some Portuguese and Brazilian friends for feedback on this too.